rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -1,122 +1,317 @@
+                event.extra_data = {'market': market}
+                event.notes = f'Deposit {event.amount} {token_0.symbol} to Pendle'
+
+            elif (
+                    event.amount == amount_1 and
+                    event.asset == token_1 and
+                    event.event_type == HistoryEventType.RECEIVE and
                     event.event_subtype == HistoryEventSubType.NONE
-                ):
-                    event.counterparty = CPT_PENDLE
-                    event.event_subtype = HistoryEventSubType.FEE
-                    event.amount = (amount := event.amount if refund_event is None else event.amount - refund_event.amount)  # noqa: E501
-                    event.notes = f'Pay {amount} ETH as vePendle state broadcast fee'
-
-            else:
-                log.error(f'Could not find pendle lock transfer for transaction {context.transaction}')  # noqa: E501
-
-            if refund_event is not None:
-                context.decoded_events.remove(refund_event)
-
-        elif context.tx_log.topics[0] == WITHDRAW_TOPIC:
-            amount = token_normalized_value_decimals(
-                token_amount=int.from_bytes(context.tx_log.data[:32]),
-                token_decimals=DEFAULT_TOKEN_DECIMALS,
-            )
-            for event in context.decoded_events:
-                if (
-                    event.amount == amount and
-                    event.address == self.ve_pendle_contract and
+            ):  # buying pt/yt
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {amount_1} {token_1.symbol} from depositing into Pendle'
+
+            elif (
+                    event.asset == token_0 and
+                    event.amount == amount_0 and
                     event.event_type == HistoryEventType.RECEIVE and
                     event.event_subtype == HistoryEventSubType.NONE
-                ):
-                    event.counterparty = CPT_PENDLE
-                    event.event_type = HistoryEventType.WITHDRAWAL
-                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
-                    event.notes = f'Withdraw {amount} PENDLE from vote escrow'
-                    break
-            else:
-                log.error(f'Could not find pendle unlock transfer for transaction {context.transaction}')  # noqa: E501
+            ):  # selling pt/yt
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Pendle'  # noqa: E501
+
+            elif (
+                    event.amount == amount_1 and
+                    event.asset == token_1 and
```

Comment:
in this if/elif thing you are checking twice for `amount/asset == amount_{i}, token_{i}` can you do something like

```
if event.amount == amount_1 and event.asset == token_1:
    if number_1:
       ...
    if number_2:
elif event.amount == amount_0 and event.asset == token_0:
     the other two ifs
```

This way you save checks